csharp/ql/lib/semmle/code/csharp/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -246,42 +246,61 @@ module Public {
       none()
     }
 
+    /**
+     * Holds if there exists a generated summary that applies to `this`.
+     */
+    final predicate hasGeneratedModel() {
+      exists(Provenance p | p.isGenerated() and this.hasProvenance(p))
+    }
+
     /**
      * Holds if all the summaries that apply to `this` are auto generated and not manually created.
+     * That is, only apply generated models, when there are no manual models.
      */
-    final predicate isAutoGenerated() {
-      this.hasProvenance(["generated", "ai-generated"]) and not this.isManual()
+    final predicate applyGeneratedModel() {
+      this.hasGeneratedModel() and
+      not this.hasManualModel()
     }
 
     /**
      * Holds if there exists a manual summary that applies to `this`.
      */
-    final predicate isManual() { this.hasProvenance("manual") }
+    final predicate hasManualModel() {
+      exists(Provenance p | p.isManual() and this.hasProvenance(p))
+    }
+
+    /**
+     * Holds if there exists a manual summary that applies to `this`.
+     * Always apply manual models if they exist.
+     */
+    final predicate applyManualModel() { this.hasManualModel() }
 
     /**
      * Holds if there exists a summary that applies to `this` that has provenance `provenance`.
      */
-    predicate hasProvenance(string provenance) { none() }
+    predicate hasProvenance(Provenance provenance) { provenance = "manual" }
```

Comment:
`provenance.isManual()`

Reply:
I think the provenance default for a summarized callable should only be "manual" and not all the manual variants.